tests/integ/modin/frame/test_equals.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import modin.pandas as pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
+
+
+@pytest.mark.parametrize(
+    "lhs, rhs, expected",
+    [
+        ([1, 2, 3], [1, 2, 3], True),
+        ([1, 2, 3], [1, 2, 4], False),  # different values
+        ([1, 2, None], [1, 2, None], True),  # nulls are considered equal
+        ([1, 2, 3], [1.0, 2.0, 3.0], False),  # float and integer types are not equal
+        ([1, 2, 3], ["1", "2", "3"], False),  # integer and string types are not equal
+    ],
+)
+@sql_count_checker(query_count=7, join_count=3)
+def test_equals(lhs, rhs, expected):
+    df1 = pd.DataFrame({"A": lhs})
+    df2 = pd.DataFrame({"A": rhs})
+    assert df1.equals(df2) == expected
+    assert df1.to_pandas().equals(df2.to_pandas()) == expected
```

Comment:
Can you construct the native frame first and then do `df1 = pd.DataFrame(native_frame)` so the query count is lower?

Also, can you add a comment explaining where the query/join count comes from?

Reply:
Done